Sources/Logging/TestingLogHandler.swift

```diff
@@ -0,0 +1,298 @@
+import Foundation
+
+#if os(Linux)
+    import Glibc
+#else
+    import Darwin.C
+#endif
+
+public struct LogMessage: CustomStringConvertible {
+    private static let timestampFormat = createTimestampFormat()
+
+    private static func createTimestampFormat() -> DateFormatter {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+        return dateFormatter
+    }
+
+    public let timestamp: Date
+    public let label: String
+
+    public let level: Logger.Level
+    public let message: Logger.Message
+    public let metadata: Logger.Metadata
+    public let source: String
+
+    public let file: String
+    public let function: String
+    public let line: UInt
```

Comment:
Can we make all of those `var`s. Since this is a `struct` this should be fine.

Reply:
Wouldn't that mean this is no longer implementing the `Sendable` implicit protocol? ...or is that only `class`es that have that requirement?

Reply:
Correct, since this is a `struct` we can safely make everything mutable and still be `Sendable` since the compiler will copy the struct when mutated

Reply:
Done